### ISSUE TITLE
panic: close of closed channel

### ISSUE THREAD

**Issue by rasky** [klauspost/pgzip#4]
While using pgzip, I'm getting this error in some (not yet fully debugged) situations:

```
panic: close of closed channel

goroutine 2364 [running]:
github.com/klauspost/pgzip.(*Writer).checkError(0xc208060100, 0x0, 0x0)
/home/ubuntu/.go_workspace/src/github.com/klauspost/pgzip/gzip.go:254 +0xee
github.com/klauspost/pgzip.(*Writer).Write(0xc208060100, 0xc20827a000, 0x8000, 0x8000, 0x8000, 0x0, 0x0)
/home/ubuntu/.go_workspace/src/github.com/klauspost/pgzip/gzip.go:273 +0x6b
[...]
```

If it can help, the underlying writer for pgzip is a io.Pipe() writer, and the other end is copied into a socket, so it looks like the bug is related to the packetization of the data over the wire, since it's not fully reproducible. 

From my reading of the code, it looks like the panic is actually a failure to propagate an underlying error code, so I will just put a print there to see what error code was triggered in the first place. Meanwhile, the traceback might be enough to point you to the bug causing the panic.

**Comment by klauspost** [comment]
Exactly. I just pushed a fix. I will do a test as well.

**Comment by klauspost** [comment]
Fixed with db54d9bf4f72c2c46e80638e26ba60083defae6d and c21006ea5f609f3267383c80c6a7de6cf9b98e1 (test)
